File: web/src/api/sysDictionaryDetail.ts
import service from '@/utils/request'

export interface SysDictionaryDetail {
    ID?: number
    label?: string
    value?: number | string
    status?: boolean
    sort?: number
    sysDictionaryID?: number
}

export interface SysDictionaryDetailSearch {
    page: number
    pageSize: number
    label?: string
    value?: number | string
    status?: boolean
    sysDictionaryID?: number | string
}

// @Tags SysDictionaryDetail
// @Summary 创建SysDictionaryDetail
export const createSysDictionaryDetail = (data: SysDictionaryDetail) => {
    return service({
        url: '/api/v1/dictionary/detail/add',
        method: 'post',
        data
    })
}

// @Tags SysDictionaryDetail
// @Summary 删除SysDictionaryDetail
export const deleteSysDictionaryDetail = (data: { ID: number }) => {
    return service({
        url: '/api/v1/dictionary/detail/delete',
        method: 'post',
        data
    })
}

// @Tags SysDictionaryDetail
// @Summary 更新SysDictionaryDetail
export const updateSysDictionaryDetail = (data: SysDictionaryDetail) => {
    return service({
        url: '/api/v1/dictionary/detail/update',
        method: 'post',
        data
    })
}

// @Tags SysDictionaryDetail
// @Summary 用id查询SysDictionaryDetail
export const findSysDictionaryDetail = (params: { ID: number | string }) => {
    return service({
        url: '/api/v1/dictionary/detail/get',
        method: 'get',
        params
    })
}

// @Tags SysDictionaryDetail
// @Summary 分页获取SysDictionaryDetail列表
export const getSysDictionaryDetailList = (params: SysDictionaryDetailSearch) => {
    return service({
        url: '/api/v1/dictionary/detail/list',
        method: 'get',
        params
    })
}

// @Tags SysDictionaryDetail
// @Summary 获取层级字典详情树形结构（根据字典ID）
export const getDictionaryTreeList = (params: { sysDictionaryID: string | number }) => {
    return service({
        url: '/api/v1/dictionary/detail/tree/list',
        method: 'get',
        params
    })
}

// @Tags SysDictionaryDetail
// @Summary 获取层级字典详情树形结构（根据字典类型）
export const getDictionaryTreeListByType = (params: { dictType: string }) => {
    return service({
        url: '/api/v1/dictionary/detail/tree/type/list',
        method: 'get',
        params
    })
}

// @Tags SysDictionaryDetail
// @Summary 根据父级ID获取字典详情
export const getDictionaryDetailsByParent = (params: { parentID: string | number; includeChildren?: boolean }) => {
    return service({
        url: '/api/v1/dictionary/detail/parent/get',
        method: 'get',
        params
    })
}

// @Tags SysDictionaryDetail
// @Summary 获取字典详情的完整路径
export const getDictionaryPath = (params: { ID: string | number }) => {
    return service({
        url: '/api/v1/dictionary/detail/path/get',
        method: 'get',
        params
    })
}

File: server/router/system/sys_dictionary_detail.go
package system

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type DictionaryDetailRouter struct{}

func (s *DictionaryDetailRouter) InitSysDictionaryDetailRouter(Router *gin.RouterGroup) {
	dictionaryDetailRouter := Router.Group("dictionary/detail").Use(middleware.OperationRecord())
	{
		dictionaryDetailRouter.POST("add", dictionaryDetailApi.CreateSysDictionaryDetail) // 新建SysDictionaryDetail

		dictionaryDetailRouter.POST("delete", dictionaryDetailApi.DeleteSysDictionaryDetail) // 删除SysDictionaryDetail
		dictionaryDetailRouter.POST("update", dictionaryDetailApi.UpdateSysDictionaryDetail) // 更新SysDictionaryDetail

		dictionaryDetailRouter.GET("get", dictionaryDetailApi.FindSysDictionaryDetail)                // 根据ID获取SysDictionaryDetail
		dictionaryDetailRouter.GET("list", dictionaryDetailApi.GetSysDictionaryDetailList)            // 获取SysDictionaryDetail列表
		dictionaryDetailRouter.GET("tree/list", dictionaryDetailApi.GetDictionaryTreeList)            // 获取字典详情树形结构
		dictionaryDetailRouter.GET("tree/type/list", dictionaryDetailApi.GetDictionaryTreeListByType) // 根据字典类型获取字典详情树形结构
		dictionaryDetailRouter.GET("parent/get", dictionaryDetailApi.GetDictionaryDetailsByParent)    // 根据父级ID获取字典详情
		dictionaryDetailRouter.GET("path/get", dictionaryDetailApi.GetDictionaryPath)                 // 获取字典详情的完整路径
	}
}
